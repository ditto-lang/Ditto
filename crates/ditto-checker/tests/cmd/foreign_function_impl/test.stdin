module Test exports (..); -- `..` shouldn't expose example_impl !

-- implemented elsewhere
example = example_impl;

foreign example_impl : (Int, Float) -> Unit;
