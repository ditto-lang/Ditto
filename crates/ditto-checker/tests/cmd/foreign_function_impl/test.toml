bin.name = "ditto-checker-testbin-check-module"
args = ["foreign_function_impl.ditto"]
fs.sandbox = true
