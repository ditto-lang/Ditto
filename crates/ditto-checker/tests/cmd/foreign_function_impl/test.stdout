{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {},
    "constructors": {},
    "values": {
      "example": {
        "doc_comments": [
          "implemented elsewhere"
        ],
        "doc_position": 0,
        "value_type": {
          "type": "Function",
          "data": {
            "parameters": [
              {
                "type": "PrimConstructor",
                "data": "Int"
              },
              {
                "type": "PrimConstructor",
                "data": "Float"
              }
            ],
            "return_type": {
              "type": "PrimConstructor",
              "data": "Unit"
            }
          }
        }
      }
    }
  },
  "types": {},
  "constructors": {},
  "values": {
    "example": {
      "doc_comments": [
        "implemented elsewhere"
      ],
      "name_span": {
        "start_offset": 91,
        "end_offset": 98
      },
      "expression": {
        "expression": "ForeignVariable",
        "data": {
          "span": {
            "start_offset": 101,
            "end_offset": 113
          },
          "variable_type": {
            "type": "Function",
            "data": {
              "parameters": [
                {
                  "type": "PrimConstructor",
                  "data": "Int"
                },
                {
                  "type": "PrimConstructor",
                  "data": "Float"
                }
              ],
              "return_type": {
                "type": "PrimConstructor",
                "data": "Unit"
              }
            }
          },
          "variable": "example_impl"
        }
      }
    }
  },
  "values_toposort": [
    "example"
  ]
}